### PR TITLE
ci: add tools to automate git-flow

### PR DESCRIPTION
**Pull request by levibostian**
This PR introduces new tools + docs on the automation of the git workflow that we follow on this project. 

The tools are installed to help us automate the tasks that we do today around merging PRs and deployment. See the bot comment in this PR? That's one of the tools! 

The docs are designed to help us and future team members learn the workflow that we use for development and deployment. 

With reviewing this PR, the biggest need that it has is reading the documentation to see if it answers all of your questions in regards to how to use these tools.

### CODE REVIEW

File: .github/pull_request_template.md
Comment (ami-aman) [inline .github/pull_request_template.md:5]: When will it be merged ? Immediately or after a certain period of time ?

File: docs/dev-notes/DEVELOPMENT.md
Comment (ami-aman) [inline docs/dev-notes/DEVELOPMENT.md:40]: Isn't it a good idea to merge line #36 and #40 ?

File: .github/pull_request_template.md
Comment (levibostian) [reply .github/pull_request_template.md:5]: Immediately. A bot will merge the request for you. 

You know how today when we squash merge pull requests, we need to remember to delete the github autogenerated description? And sometimes github has made the squashed commit *not* the title of the pull request like we want? By having a bot merge our PR, the bot will do all of this for us so we don't need to remember to do anything!

File: docs/dev-notes/DEVELOPMENT.md
Comment (levibostian) [reply docs/dev-notes/DEVELOPMENT.md]: I removed the deployment section 👍

File: .github/pull_request_template.md
Comment (ami-aman) [reply .github/pull_request_template.md:5]: I like this auto-botic merging. Can you mention that `it will automatically be merged as soon as you add the tag`
When will it be merged ? This question popped up in my mind as I read. I feel other devs might also feel the same.

File: docs/dev-notes/DEVELOPMENT.md
Comment (ami-aman) [reply docs/dev-notes/DEVELOPMENT.md]: Thanks. Resolving conversation for you.

File: .github/pull_request_template.md
Comment (levibostian) [reply .github/pull_request_template.md]: Thanks for bringing that up. 

I added a note about a bot merging the PR. Does the change look good to you?

I wanted to avoid using the word "immediately" because it's not immediate but more like 15 seconds until the bot merges it.

Comment (ami-aman) [reply .github/pull_request_template.md]: This looks perfect 👍

**Review by ami-aman — APPROVED** [review]
👍

File: .github/workflows/deploy-git-tag.yml
Comment (joepurdy) [inline .github/workflows/deploy-git-tag.yml:12]: I think it's largely fine that we're depending on semantic-release plugins you've authored. I just want to caution that we'll want to make sure to support this work so that you (@levibostian) aren't the only engineer that can contribute fixes. There's still of course the standard OSS workflow of "fork it, open a PR from your fork, get it merged" so there's not a huge risk here. But if it's something Customer.io's teams are the primary consumers/maintainers of we might benefit from reducing the added friction of forks by having additional maintainers to the source repository (wherever they live)

File: .github/workflows/pr-helper.yml
Comment (joepurdy) [inline .github/workflows/pr-helper.yml:13]: Same point as before about external dependencies maintained by internal users

File: .github/workflows/promote.yml
Comment (joepurdy) [inline .github/workflows/promote.yml:12]: Same point as before about external dependencies maintained by internal users

**Review by joepurdy — APPROVED** [review]
LGTM!

Left some comments that echo some discussions we've had in private. Overall, this tracks to me and I appreciate the detail and thought that went into describing the workflow for newcomers.

Would be great to prepare an engineering talk for the mobile team (and any other engineers who are curious) about how you've solved this workflow using semantic-release and the custom workflow plugins. I'd be happy to help with that if you'd like!

**Pull request merged**
